app/Http/Controllers/Admin/NotificationAdminController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\BaseController;
+use App\Http\Requests\NotificationRequest;
+use App\Models\Controller\ControllerPosition;
+use App\Models\Notification\Notification;
+use Carbon\Carbon;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class NotificationAdminController extends BaseController
+{
+    /**
+     * Get a list of all notifications, optionally including expired
+     * notifications via query parameter in request.
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function getNotifications(Request $request) : JsonResponse
+    {
+        // if the requester wishes to view expired notifications
+        // include expired in the query results
+        if ($request->query('include_expired', false)) {
+            $notifications = Notification::withCount('controllers')->get();
+        } else {
+            $notifications = Notification::active()->withCount('controllers')->get();
+        }
+
+        return response()->json(['notifications' => $notifications]);
+    }
+
+    /**
+     * Retrieve the details on a specified notification, appending
+     * relevant details for further consumption
+     *
+     * @param Notification $notification
+     * @return JsonResponse
+     */
+    public function getNotification(Notification $notification) : JsonResponse
+    {
+        // eager load controllers relationship
+        $notification->load('controllers');
```

Comment:
As the model is loaded prior to the controller and we call the relationship a few lines down, I don't think eager loading does anything significant, in this instance

Reply:
Calling the relationship without eager loading it first would result in an N + 1, wouldn't it?